apps/cowswap-frontend/src/modules/swap/hooks/useSwapButtonContext.ts

```diff
@@ -128,6 +128,9 @@ export function useSwapButtonContext(input: SwapButtonInput): SwapButtonsContext
       openNativeWrapModal()
     },
     openSwapConfirm() {
+      // TODO: move permit info and status to an atom
+      // We need to know before opening the confirmation modal if there's a cached permit
+      // and if it's valid to know what to show...
```

Comment:
During testing I realized that it doesn't matter for this, and it makes for a nice UI as is.

When there's a cached permit that is still valid but wasn't executed yet, we just don't request the signature and show this step as complete.

We'll need to move permit info to an atom, but it does not need to be consumed here.

```suggestion
```